### ISSUE TITLE
Bug: codeigniter4 PHP 8.2 session error

### ISSUE THREAD

**Issue by kimdoyeonsa** [codeigniter4/CodeIgniter4#7122]
### PHP Version

8.2

### CodeIgniter4 Version

4.2.12

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

apache

### Database

MariaDB 10.10

### What happened?

php8.2

apache2
err_empty_response

nginx 
ini_set(): A session is active

### Steps to Reproduce

php 8.2 apt install
server apache2,nginx (php8.2-fpm)
session error for linux
$this->session = \Config\Services::session();

### Expected Output

Hope the session bug goes away

### Anything else?

[
![2023-01-15_124504](https://user-images.githubusercontent.com/48858134/212522876-6668dee2-997c-40fe-add4-f2bded9df801.jpg)
](url)

**Comment by kenjis** [comment]
> CodeIgniter4 Version
php 8.2 fix version

Please provide CodeIgniter4 version you use.

**Comment by kenjis** [comment]
Please try Local Development Server:
https://codeigniter4.github.io/CodeIgniter4/installation/running.html#local-development-server

If it works, the issue is in your web server configuration.

**Comment by kimdoyeonsa** [comment]
4.3.1

**Comment by kimdoyeonsa** [comment]
4.2.12 higher

**Comment by kenjis** [comment]
Cannot reproduce. 

```php
<?php

namespace App\Controllers;

class Home extends BaseController
{
    public function index()
    {
        $session = \Config\Services::session();
    }
}
```
I got no error.

**Comment by kimdoyeonsa** [comment]
ubuntu apt install php8.2 version session

**Comment by kimdoyeonsa** [comment]
windows php8.2 no error

**Comment by kimdoyeonsa** [comment]
nginx ini_set error session.save_path

**Comment by kimdoyeonsa** [comment]
namespace App\Controllers;

class Home extends BaseController
{
    public function index()
    {
        $session = \Config\Services::session();
    }
}

<-- nginx,apache php 8.2 session nginx 500error

php 8.1 no error

nginx error log

recv() failed (104: Connection reset by peer) while reading response header
from upstream, client server: localhost, request: "GET / HTTP/1.1",
upstream: "fastcgi://unix:/run/php/php8.2-fpm.sock:"

2023년 1월 15일 (일) 오후 2:32, kenjis ***@***.***>님이 작성:

**Comment by kenjis** [comment]
> <-- nginx,apache php 8.2 session nginx 500error

What do you mean?
Could you explain plain English?

**Comment by kimdoyeonsa** [comment]
that code gives error.

2023년 1월 16일 (월) 오후 2:02, kenjis ***@***.***>님이 작성:

**Comment by kenjis** [comment]
Please provide minimum Steps that everyone can Reproduce the error.

I cannot reproduce the error with `php spark serve`.
So it seems not a framework issue, but your server configuration issue.

**Comment by kimdoyeonsa** [comment]
php 8.2 linux

$session = \Config\Services::session();

php spark serve also gives an error
Even if you install php8.2 version several times, an error occurs on the
session code side.
There seems to be an error in saving the session.

the code that works is $_SESSION.



2023년 1월 16일 (월) 오후 3:17, kenjis ***@***.***>님이 작성:

**Comment by kenjis** [comment]
What Session driver do you use?

**Comment by kimdoyeonsa** [comment]
Use the session driver in the php linux common package

2023년 1월 17일 (화) 오전 7:35, kenjis ***@***.***>님이 작성:

**Comment by kenjis** [comment]
CI4 has its own drrivers.
See https://codeigniter4.github.io/CodeIgniter4/libraries/sessions.html#session-drivers
If you don't change in the config file, you are using `FileHandler`.

**Comment by kimdoyeonsa** [comment]
FileHandler use.

2023년 1월 17일 (화) 오전 8:51, kenjis ***@***.***>님이 작성:

**Comment by kenjis** [comment]
> php spark serve also gives an error

What is the exact error message?

**Comment by kimdoyeonsa** [comment]
Unable to connect

2023년 1월 17일 (화) 오전 9:57, kenjis ***@***.***>님이 작성:

**Comment by kenjis** [comment]
> Unable to connect

Is it the error in your browser?
Is there any message in the terminal you run `php spark serve`?
Or in the CodeIgniter log files?

**Comment by kimdoyeonsa** [comment]
you will see the error message in your browser.
No error appears in the terminal window, and logs are not saved in
CodeIgniter.

2023년 1월 17일 (화) 오전 10:27, kenjis ***@***.***>님이 작성:

**Comment by kimdoyeonsa** [comment]
I installed the php8.2 package, but if only 2 are installed, the session does not work.

php8.2-xhprof php8.2-tideways

**Comment by im-machakata** [comment]
Had the same issue while trying to run my app on [Gitpod.io](//gitpod.io). Let me add my patch

**Comment by kenjis** [comment]
We cannot reproduce. Close.
